octodns/provider/ns1.py

```diff
@@ -351,12 +358,35 @@ def _data_for_dynamic_A(self, _type, record):
             for georegion in meta.get('georegion', []):
                 geos.add(self._REGION_TO_CONTINENT[georegion])
 
-            # Countries are easy enough to map, we just have ot find their
+            # Countries are easy enough to map, we just have to find their
             # continent
+            #
+            # NOTE: Special handling for Oceania
+            # NS1 doesn't support Oceania as a region. So the Oceania countries
+            # will be present in meta['country']. If all the countries in the
+            # Oceania countries list are found, set the region to OC and remove
+            # individual oceania country entries
+
+            oc_countries = []
             for country in meta.get('country', []):
-                con = country_alpha2_to_continent_code(country)
+                # country_alpha2_to_continent_code fails for Pitcairn ('PN')
+                if country == 'PN':
+                    con = 'OC'
+                else:
+                    con = country_alpha2_to_continent_code(country)
+
+                if con == 'OC':
+                    oc_countries.append(country)
                 geos.add('{}-{}'.format(con, country))
 
+            if len(oc_countries):
+                all_oc_countries = self._CONTINENT_TO_LIST_OF_COUNTRIES['OC']
+                if sorted(oc_countries) == sorted(all_oc_countries):
```

Comment:
If you change `oc_countries` and `_CONTINENT_TO_LIST_OF_COUNTRIES` to be `set`s rather than `list`s you can just compare them directly here. Should be simpler/cleaner than having to sort.